smart-contracts/contracts/cl-vault/src/state.rs

```diff
@@ -81,17 +81,24 @@ pub struct Position {
     pub claim_after: Option<u64>, // this should be off chain computed and set in order to avoid forfeiting incentives
 }
 
-pub const POSITION: Item<Position> = Item::new("position");
-
+// positions in the contract, the key should be the same as the position's id in Osmosis
+pub const POSITIONS: Map<u64, Position> = Map::new("positions");
+pub const MAIN_POSITION: Item<u64> = Item::new("main_position");
+pub const CURRENT_POSITION_ID: Item<u64> = Item::new("current_position_id");
+pub const CURRENT_CLAIM_AFTER: Item<Option<u64>> = Item::new("current_claim_after");
 pub const SHARES: Map<Addr, Uint128> = Map::new("shares");
 
-/// The merge of positions currently being executed
+/// boolean signifying whether the current executing merge is for the main position
+pub const MERGE_MAIN_POSITION: Item<bool> = Item::new("merge_main_position");
+
+/// The queue of positions currently being merged
 pub const CURRENT_MERGE: Deque<CurrentMergeWithdraw> = Deque::new("current_merge");
 
 #[cw_serde]
 pub struct CurrentMergePosition {
     pub lower_tick: i64,
     pub upper_tick: i64,
+    pub claim_after: Option<u64>,
```

Comment:
To make the unit visible in client code:
```suggestion
    pub claim_after_secs: Option<u64>,
```